components/engine_rocks/src/config.rs

```diff
@@ -262,124 +266,33 @@ impl From<BlobRunMode> for DBTitanDBBlobRunMode {
     }
 }
 
-macro_rules! rocksdb_enum_mod {
-    ($name:ident $enum:ident { $($variant:ident = $value:expr, )* }) => {
-        pub mod $name {
-            use std::fmt;
-
-            use serde::{Serializer, Deserializer};
-            use serde::de::{self, Unexpected, Visitor};
-            use rocksdb::$enum;
-            use case_macros::*;
-
-            pub fn serialize<S>(mode: &$enum, serializer: S) -> Result<S::Ok, S::Error>
-                where S: Serializer
-            {
-                serializer.serialize_i64(*mode as i64)
```

Comment:
One more thing, use `$value:expr` here, considering `DBRateLimiterMode` starts with 1.

Reply:
And add a test asserting serialized string equals original string.

Reply:
> One more thing, use $value:expr here, considering DBRateLimiterMode starts with 1.

Seems DBRateLimiterMode starts with 1 not 0 does not affect the result, i have test both use 
```
serializer.serialize_i64(*mode as i64)
```
and 
```
 match mode {
         $( $enum::$variant => serializer.serialize_i64($value as i64), )*
  }
```
Test code is :

```
        let x = r#"
            rate-limiter-mode = "read-only"
        "#;

        let config: DbConfig = toml::from_str(x).unwrap();
        assert_eq!(config.rate_limiter_mode, DBRateLimiterMode::ReadOnly);

        println!("{:?}", toml::to_string(&config).unwrap());
        assert!(
            toml::to_string(&config)
                .unwrap()
                .contains("rate-limiter-mode = 1")
        );
```

Both can pass, Still, shall we use to latter form anyhow?


> And add a test asserting serialized string equals original string.

I can't catch your point. Currently any config value is serialized as numeric value not matter what the input format is, what's the meaning of  `serialized string`?

Reply:
Oh, it's because rust-rocksdb codes it from 1 too: https://github.com/tikv/rust-rocksdb/blob/06aa625e21b0b2c632cc02b7d001e6d5671bbc38/librocksdb_sys/src/lib.rs#L379. Regardless of that, using `$value` is better.

And I mean testing `toml::to_string(toml::from_str(str)) == str`, where str is "rate-limiter-mode = 1, ..."

Reply:
>  And I mean testing toml::to_string(toml::from_str(str)) == str, where str is "rate-limiter-mode = 1, ..." 

The test `test_serde_custom_tikv_config` in integration test has cover the case , but seems that `make test` will not trigger integration test. anyway, i wiil add some test about `rate-limiter-mode`